### ISSUE TITLE
Sails does not apply population criteria

### ISSUE THREAD

**Issue by glenswift** [balderdashy/sails#2704]
I have model `User`. To simplify the issue it has not attributes (except id that is produced by sails engine):

    module.exports = {

    }

I have also model `UserPost`:

    module.exports = {
        attributes: {
            user: {
                required: true,
                model: 'user'
            }
        }
    }

I want to find posts of user with id equal to 42. I have that records in database, both user with id 42, and its posts.

This doesn't work:

    UserPost.find().populate('user', {
        id: 42
    }).exec(function(err, posts) {
        res.json(posts);
    });

`posts` is array of posts of all users.

Is it expected bahavior?

**Comment by particlebanana** [comment]
So in the query:

```javascript
UserPost.find().populate('user', { id: 42 }).exec(/**/)
```

You are saying give me all the `UserPost` records and populate the `user` associations on only the `UserPost` records who are associated with user 42.

If you change it to:

```javascript
UserPost.find({ user: 42 }).populate('user').exec(/**/)
```

This query changes to give me all the `UserPost` records for userId 42 and populate the user account.

We don't currently support sub queries so you couldn't say give me all the `UserPost` records for any users who have signed up in the last day. You would first have to find the users and then use that to get the userposts.

**Comment by glenswift** [comment]
Well, it seems like I've explained my problem not clear. Yes, there is no problem to put `id` exactly to `find()` query. The problem appears as soon as I want to work with something else instead of it. Something like: give me all the `UserPost` records for user with given `username` (as an example).

In the documentation I've found very [similar example ][0] so this probably should work. Either with `id` population or `username` population or any attribute from `User` model. Or not?

[0]: https://github.com/balderdashy/sails-docs/blob/8fc2694a795bc753277f9b970b835dbb384ebfbe/reference/waterline/queries/populate.md#example-usage
